addons/web/controllers/main.py

```diff
@@ -822,22 +826,17 @@ def _write_group_header(self, row, column, label, group, group_depth=0):
 
             decimal_places = [res['decimal_places'] for res in group._model.env['res.currency'].search_read([], ['decimal_places'])]
             decimal_places = max(decimal_places) if decimal_places else 2
+            monetary_format = '0' + self.decimal_separator + decimal_places * '0'
```

Comment:
We could improve this part by defining the monetary format where you define `self.float_format`
By using the company_id in the `request.env` you can retrieve the company `currency_id` and thus the correct number of digits for the monetary format

Reply:
There's an issue with using the currency of the company as it may not be the currency used in the fields we are trying to print (see `test_float_representation`). I moved this part in __init__ and it still uses the same approach of printing with the biggest decimal precision set on currencies (I'm not sure we can find the currency of the field we are writing)

Reply:
I see, maybe I'm wrong but we cannot retrieve the currency of the field, so this seems a limitation to me

Reply:
I see, maybe I'm wrong but we cannot retrieve the currency of the field, so this seems a limitation to me

(Apparently github didn't like this comment and it was lost, I just rewrote it)